src/ripple/app/tx/impl/Transactor.h

```diff
@@ -98,32 +94,29 @@ class Transactor
     explicit
     Transactor (ApplyContext& ctx);
 
-    void calculateFee ();
-
-    // VFALCO This is the equivalent of dynamic_cast
-    //        to discover the type of the derived class,
-    //        and therefore bad.
-    virtual
-    bool
-    mustHaveValidAccount()
-    {
-        return true;
-    }
-
     // Returns the fee, not scaled for load (Should be in fee units. FIXME)
     virtual std::uint64_t calculateBaseFee ();
 
     virtual void preCompute();
-    virtual TER checkSeq ();
-    virtual TER payFee ();
-    virtual TER checkSign ();
+
     virtual TER doApply () = 0;
 
 private:
+    TER checkSeq ();
+    TER checkSign ();
+    TER payFee ();
     TER checkSingleSign ();
     TER checkMultiSign ();
 };
 
+/** Performs early sanity checks on the account and fee fields */
+TER
+preflight1 (PreflightContext const& ctx);
+
+/** Checks whether the signature appears valid */
+TER
+preflight2 (PreflightContext const& ctx);
+
```

Comment:
I like that you moved `Transactor::preflight` to a free function, but I'm not loving the function being split.
Pros: You get the simple account and fee checks out of the way before any possible more expensive checks in the transaction-specific `preflight` function, and the even more expensive signature check.
Cons: Now we've got two functions to maintain, and we end up with unfortunate situations like `CancelTicket::preflight` and `SusPayCancel::preflight`, where both functions are called without anything in between, adding to complication.

Since you've made the change, I assume you've decided it's worth it, so I'll offer a couple of suggestions:
1. Consider adding a combined function that can be called from `CancelTicket` and any other places that don't need to do anything in between. It could be back in `Transactor::preflight`, or another free function.
2. Consider renaming the functions so it's at least clear why there are two functions that most transactors need to call. Suggestions:
  * `preFlightAccountFee` / `preFlightSignature`
  * `preFlightBefore` / `preFlightAfter`
  * `preFlightCheap` / `preFlightExpensive`.